### PR TITLE
oauth2 against liquid users app

### PR DESCRIPTION
**Pull request by mgax**

### CODE REVIEW

File: hoover/contrib/__init__.py [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 class installed:
     twofactor = ('hoover.contrib.twofactor' in settings.INSTALLED_APPS)
+    oauth2 = ('hoover.contrib.oauth2' in settings.INSTALLED_APPS)
```
Comment (gabriel-v) [inline hoover/contrib/__init__.py]: Unrelated aspect: one would have to either modify `common.py` or overwrite it in `local.py`. Modifying `common.py` for each instance causes `git` to complain.

How do we document this problem? Do we invite users to copy/paste the whole `INSTALLED_APPS` list to `local.py`?

File: hoover/contrib/oauth2/Readme.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+### setup
+* add `hoover.contrib.oauth2` to `INSTALLED_APPS`
+* set `HOOVER_OAUTH_LIQUID_URL` to the URL of the liquid users app
+* set `HOOVER_OAUTH_LIQUID_CLIENT_ID` and `HOOVER_OAUTH_LIQUID_CLIENT_SECRET`
```
Comment (gabriel-v) [inline hoover/contrib/oauth2/Readme.md]: Where can one get the `HOOVER_OAUTH_LIQUID_CLIENT_ID` and `HOOVER_OAUTH_LIQUID_CLIENT_SECRET` keys?
It'd be really nice to have the documentation here a bit more complete and have link to the [liquid-core documentation](https://github.com/hoover/liquid-core/tree/master/liquidcore/site)? (when you'll write it, that is)

At least a reference to the `liquid-code / Applications / Authentication` admin would be nice.

File: hoover/contrib/oauth2/views.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    token_resp = requests.post(
+        token_url,
+        data={
+            'redirect_uri': redirect_uri,
+            'grant_type': 'authorization_code',
+            'code': code,
+        },
+        auth=(LIQUID_CLIENT_ID, LIQUID_CLIENT_SECRET),
+    )
+    assert token_resp.status_code == 200
```
Comment (gabriel-v) [inline hoover/contrib/oauth2/views.py]: Could we have a more comprehensive error message show up in the logs? 

The assert doesn't really help future debugging efforts.
Maybe even log the response status code, so one could investigate further.

File: hoover/contrib/oauth2/views.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    access_token = token_data['access_token']
+    token_type = token_data['token_type']
+    assert token_type == 'Bearer'
+    refresh_token = token_data['refresh_token']
+    profile_url = LIQUID_URL + '/accounts/profile'
+    profile_resp = requests.get(
+        profile_url,
+        headers={'Authorization': 'Bearer {}'.format(access_token)},
+    )
+    assert profile_resp.status_code == 200
```
Comment (gabriel-v) [inline hoover/contrib/oauth2/views.py]: Same as above, an error message would be really helpful.

File: hoover/site/urls.py [hunk 5]
```diff
@@ -22,10 +22,23 @@
         url(r'^invitation/(?P<code>.*)$', twofactor_views.invitation),
         url(r'^accounts/login/$', auth_views.login, kwargs={
             'authentication_form': twofactor_views.AuthenticationForm}),
+        url(r'^accounts/', include('django.contrib.auth.urls')),
+    ]
+
+elif installed.oauth2:
```
Comment (gabriel-v) [inline hoover/site/urls.py]: Could we somehow fool-proof the case when the admin has configured both oauth2 AND twofactor auth?

Maybe throw an exception if both are installed, so it the server even start?

Also, we should probably document somewhere that these are mutually exclusive, even if it's obvious after you think about it.

File: hoover/contrib/__init__.py (hunk 1 above)
Comment (mgax) [inline hoover/contrib/__init__.py]: Or import the list from `common.py` and append to it? That seems cleaner.

**Pull request merged**